#164797136 Configure Authors Haven to use Postgres

NedyUdombat opened the pull request:
#### What does this PR do?
Configure Authors haven to use postgres as its database engine

#### Description of Task to be completed?
* uninstall mongo and it's dependencies
* install postgres and its dependencies (pg, pg-hstore)
* install babel and its dependencies
* setup babel configuration file
* install sequelize and its dependencies
* setup sequelize configuration file
* initialise Authors Haven to use sequelize
* create a database connnection point to sequelize (`server/config/config.js`)
* create a template environment file to store database connection details.
* remove all old models and confgurations that were based on mongo database.

#### How should this be manually tested?
* clone `https://github.com/andela/zinnia-ah-backend.git`
* checkout to  a new branch = `ch-setup-postgres-164797136`
* pull from the remote branch to update the local branch
* run `npm install`
* run `sudo npm install -g pg`
* run `cp .env.example .env`
* create a database
* fill in the details in your `.env`
* run `sequelize db:migrate`

#### Any background context you want to provide?
None 

#### What are the relevant pivotal tracker stories?
#164797136 

#### Screenshots (if appropriate)
None

#### Questions:
None.

Commit:
[chore #164797136]Configure Authors Haven to use Postgres

Commit:
[chore #164797136]Configure Authors Haven to use Postgres

Commit:
[chore #164797136]Configure Authors Haven to use Postgres

Commit:
[chore #164797136]Configure Authors Haven to use Postgres

Commit:
[chore #164797136]Setup migration for database

Commit:
[chore #164797136]Setup migration for database

Commit:
[chore #164797136]Setup migration for database

Commit:
[chore #164797136]Setup migration for database

Commit:
[chore #164797136]Correct typographical error in variable naming in user model

Commit:
[chore #164797136]Correct typographical error in variable naming in user model

Commit:
[chore #164797136]Correct typographical error in variable naming in user model

Commit:
[chore #164797136]Correct typographical error in variable naming in user model

emmsdan reviewed (dismissed):
you created a .babelrc file and installed babel dependencies but you have not configured the *scripts* to run it. i.e `npm run build`

NedyUdombat commented:
> you created a .babelrc file and installed babel dependencies but you have not configured the _scripts_ to run it. i.e `npm run build`

This was to help compile sequelize to run, `npm run build` will be setup when deploying the base app to heroku

Japhethca commented on .babelrc:
```diff
@@ -0,0 +1,3 @@
+{
+    "presets": ["env"]
+  }
```
fix alignment

Japhethca commented on server/config/config.js:
```diff
@@ -0,0 +1,15 @@
+
+dotenv.config();
+
+module.exports = {
```
let use es6 export

Japhethca commented on server/migrations/20190325172457-create-users.js:
```diff
@@ -0,0 +1,32 @@
+module.exports = {
```
same as above

Japhethca commented on server/models/index.js:
```diff
@@ -0,0 +1,36 @@
+
```
same as above

Commit:
chore(root, server) Implement PR feedback

properly align contents of .babelrc file
rename model and migration files to foloow proper naming coventions
fix merge conflicts while rebasing

[finish #164797136]

Commit:
chore(root, server) Implement PR feedback

properly align contents of .babelrc file
rename model and migration files to foloow proper naming coventions
fix merge conflicts while rebasing

[finish #164797136]

Commit:
chore(server)Fix merge conflict

Commit:
[chore #164797136]Fix merge conflicts with base branch

Japhethca committed:
chore(housekeeping):restructure codebase

Japhethca committed:
chore(housekeeping): get latest changes on develop

Japhethca committed:
add file to middlewares folder

Japhethca committed:
#7 restructure-codebase

Commit:
[chore #164797136](root, src) Rebase todefault branch

Commit:
[chore #164797136](root) Fix merge conflict remote branch
